src/core/src/preprocess/pre_post_process.cpp

```diff
@@ -62,7 +61,25 @@ struct RTInfoCache {
         });
     }
 
-    std::unordered_map<ov::Node*, ov::RTMap> m_rt_info_cache;
+private:
+    struct CacheKey {
+        explicit CacheKey(const std::shared_ptr<ov::Node>& n) : hash_value{std::hash<ov::Node*>{}(n.get())}, node{n} {}
+        struct Hash {
+            size_t operator()(const CacheKey& k) const {
+                return k.hash_value;
+            }
+        };
+        struct Equal {
+            bool operator()(const CacheKey& lhs, const CacheKey& rhs) const {
+                return !lhs.node.owner_before(rhs.node) && !rhs.node.owner_before(lhs.node);
+            }
+        };
+
+    private:
+        size_t hash_value{0};
+        std::weak_ptr<ov::Node> node;
```

Comment:
```suggestion
        size_t hash_value{};
        std::weak_ptr<ov::Node> node{};
```
Or no initialization,
Block creation with default ctor?

Reply:
There's only one `CacheKey` constructor defined which enforces non-default initialization 
(zero defaulted hash is to avoid static analyzers false-positive hit).
It's also private and quite small structure, so I think it need not be heavily protected.